daemon/container_unix.go

```diff
@@ -914,6 +914,12 @@ func (container *Container) allocateNetwork() error {
 		if mode.IsDefault() {
 			networkName = controller.Config().Daemon.DefaultNetwork
 		}
+		if mode.IsUserDefined() {
+			if n, err := container.daemon.FindNetwork(networkName); err == nil {
+				networkName = n.Name()
```

Comment:
I understand the intent. but not quite sure I understand the reason on how having the network-id as the name will cause the container to make a duplicate attempt to connect to the same network. Could you pls help me understand that ?